components/sidebar/sidebar.jsx

```diff
@@ -478,16 +490,82 @@ export default class Sidebar extends React.PureComponent {
         );
     }
 
+    renderOrderedChannels = () => {
+        const {orderedChannelIds} = this.props;
+
+        const channelsToHide = ['unreads', 'favorite'];
+
+        const orderedChannelSections = orderedChannelIds.map((section) => {
+            return {
+                type: section.type,
+                name: section.name,
+                items: section.items.map(this.createSidebarChannel),
+            };
+        });
+
+        return (
+            <Scrollbars
+                ref='scrollbar'
+                autoHide={true}
+                autoHideTimeout={500}
+                autoHideDuration={500}
+                renderThumbHorizontal={renderThumbHorizontal}
+                renderThumbVertical={renderThumbVertical}
+                renderView={renderView}
+                onScroll={this.onScroll}
+                style={{position: 'absolute'}}
+            >
+                <div
+                    id='sidebarChannelContainer'
+                    className='nav-pills__container'
+                >
+                    {orderedChannelSections.map((section) => {
+                        if (channelsToHide.indexOf(section.type) !== -1 && section.items.length === 0) {
+                            return null;
+                        }
+
+                        const sectionId = `${section.type}Channel`;
+
+                        return (
+                            <ul
+                                key={section.type}
+                                className='nav nav-pills nav-stacked'
+                            >
+                                <li>
+                                    <h4 id={sectionId}>
+                                        <ChannelName
+                                            channelType={section.type}
+                                            channelName={section.name}
+                                            browsePublicDirectChannels={this.showMorePublicDirectChannelsModal}
+                                        />
+                                        <ChannelCreate
```

Comment:
Despite my initial confusion about what that component does, I really like how it allows us to just loop through the sections without having to create each one manually